src/theory/bv/theory_bv.cpp

```diff
@@ -282,6 +282,12 @@ TrustNode TheoryBV::ppRewrite(TNode t, std::vector<SkolemLemma>& lems)
       res = RewriteRule<ZeroExtendEqConst>::run<false>(t);
     }
   }
+  // When int-blasting, it is better to handle most overflow operators
+  // natively, rather than to eliminate them eagerly.
+  if (options().smt.solveBVAsInt == options::SolveBVAsIntMode::OFF)
+  {
+    res = d_rewriter.eliminateOverflows(res);
+  }
```

Comment:
I'm not sure this would cover all the cases to eliminate overflow predicates since this is now only done in ppRewrite(). If we now internally create a new node with overflow predicates rewrite() would not ensure to eliminate them anymore, but we'd need to additionally call eliminateOverflows().

Reply:
That's a good point. On the other hand, if we keep it in the rewriter, then these are eliminated by apply-substs, which happens before int-blasting.

@ajreynol do you have an idea on how to overcome this issue of overflow nodes that are created internally? 

Alternatively, maybe we can move int-blasting to happen before apply-substs?